apps/condo/domains/billing/schema/ValidateQRCodeService.test.js

```diff
@@ -0,0 +1,239 @@
+                    path: ['result'],
+                    extensions: {
+                        mutation: 'validateQRCode',
+                        code: 'INTERNAL_ERROR',
+                        type: 'NOT_FOUND',
+                        message: 'Organization with provided TIN is not registered with Doma.ai',
+                    },
+                }])
+            })
+        })
+
+        test('should throw if no AcquiringIntegrationContext was found', async () => {
+            const [integration] = await createTestAcquiringIntegration(adminClient)
+            const [anotherOrganization] = await createTestOrganization(adminClient)
+            const [acquiringContext] = await createTestAcquiringIntegrationContext(adminClient, anotherOrganization, integration)
+            await catchErrorFrom(async () => {
+                await validateQRCodeByTestClient(adminClient, { qrCode: qrCodeString })
+            }, ({ errors }) => {
+
+                expect(errors).toMatchObject([{
+                    message: 'Organization with provided TIN does not have an active AcquiringIntegrationContext',
+                    path: ['result'],
+                    extensions: {
+                        mutation: 'validateQRCode',
+                        code: 'INTERNAL_ERROR',
+                        type: 'NOT_FOUND',
+                        message: 'Organization with provided TIN does not have an active AcquiringIntegrationContext',
+                    },
+                }])
+            })
+            await updateTestAcquiringIntegrationContext(adminClient, acquiringContext.id, { deletedAt: faker.date.past() })
+        })
+
+        test('should throw if found AcquiringIntegrationContext is not active (status !== Finished)', async () => {
+            const [integration] = await createTestAcquiringIntegration(adminClient)
+            const [acquiringContext] = await createTestAcquiringIntegrationContext(adminClient, organization, integration, { status: 'InProgress' })
+            await catchErrorFrom(async () => {
+                await validateQRCodeByTestClient(adminClient, { qrCode: qrCodeString })
+            }, ({ errors }) => {
+
+                expect(errors).toMatchObject([{
+                    message: 'Organization with provided TIN does not have an active AcquiringIntegrationContext',
+                    path: ['result'],
+                    extensions: {
+                        mutation: 'validateQRCode',
+                        code: 'INTERNAL_ERROR',
+                        type: 'NOT_FOUND',
+                        message: 'Organization with provided TIN does not have an active AcquiringIntegrationContext',
+                    },
+                }])
+            })
+            await updateTestAcquiringIntegrationContext(adminClient, acquiringContext.id, { deletedAt: faker.date.past() })
+        })
+    })
+
+    test('should throw limit exceeded error on too many calls', async () => {
```

Comment:
There is no need to make all this requests. You can set counter to MAX_CLIENT_VALIDATE_QR_CODE_BY_WINDOW - 1 and execute the code

Reply:
I'd like to be sure that all previous requests work fine